src/App.js

```diff
@@ -165,19 +162,32 @@ function App() {
     [guesses, word]
   );
 
+  const latestMatchCounts = useMemo(() => {
+    const allResults = letters.map((l) => matches[l]);
+    const newMatches = allResults.filter(x => x==="match").length - foundMatches;
+    setFoundMatches(foundMatches + newMatches);
+    const newMisMatches = allResults.filter(x => x==="mismatch").length;
+    console.log(newMatches, newMisMatches);
+    return {newMatches, newMisMatches};
+  }, [matches]);
+
   const message = useMemo(() => {
-    const r = Math.floor(Math.random() * 2);
-    if (!guesses.length) return welcome;
-    if (newMatches === 1 && !done) return newMatched1[r];
-    if (newMatches === 2 && !done) return newMatched2[r];
-    if (newMatches === 3 && !done) return newMatched3[r];
-    if (newMisMatches === 1 && !done) return newMisMatch1[r];
-    if (newMisMatches === 2 && !done) return newMisMatch2[r];
-    if (!done) return NoNew[r];
-    if (done && giveup === false) return winner[0] + guesses.length + winner[1];
-    if (done && giveup === true) return loser[0] + String(guesses.length-1) + loser[1];
+    const randomMessage = (messages) => messages[Math.floor(Math.random() * messages.length)];
+    if (!guesses.length) return randomMessage(messages.welcome);
+    if (latestMatchCounts.newMatches === 1 && !done) return randomMessage(messages.newMatched1);
+    if (latestMatchCounts.newMatches === 2 && !done) return randomMessage(messages.newMatched2);
+    if (latestMatchCounts.newMatches === 3 && !done) return randomMessage(messages.newMatched3);
+    if (latestMatchCounts.newMisMatches === 1 && !done) return randomMessage(messages.newMisMatch1);
+    if (latestMatchCounts.newMisMatches === 2 && !done) return randomMessage(messages.newMisMatch2);
+    if (!done) return randomMessage(messages.NoNew);
+    if (done && giveup === false) return messages.winner[0] + guesses.length + messages.winner[1];
+    if (done && giveup === true) return messages.loser[0] + String(guesses.length-1) + messages.loser[1];
     return "";
+<<<<<<< HEAD
```

Comment:
Looks like merge hasn't been completed here.  Probably need to delete the "<<<<", "=====", ">>>>>" lines and line 189.

Reply:
Not sure why it did that.